ClangSharpPInvokeGenerator/Program.cs

```diff
@@ -158,31 +158,38 @@ public static void Main(string[] args)
                 sw.WriteLine("    using System.Runtime.InteropServices;");
                 sw.WriteLine();
 
-                var structVisitor = new StructVisitor(sw);
+                var structDeclWriter = new CursorWriter(sw, indentation: 1, (cursor) => cursor.Kind == CXCursorKind.CXCursor_StructDecl);
```

Comment:
Could potentially expose a method that takes the predicate, rather than the constructor, so you can reuse the instance across multiple writes...